app/models/github_user.rb

```diff
@@ -11,6 +11,11 @@ def accept_repository_invitation(invitation_id, **options)
     end
   end
 
+  def watch_repository(repo)
+    options = { subscribed: true }
+    @client.update_subscription(repo.github_repository.full_name, options)
```

Comment:
I think writing this as `@client.update_subscription(repo.github_repository.full_name, subscribed: true)` is a little neater